spark/src/main/java/org/apache/iceberg/actions/ExpireSnapshotsAction.java

```diff
@@ -210,7 +210,8 @@ public ExpireSnapshotsAction deleteWith(Consumer<String> newDeleteFunc) {
   }
 
   @Override
-  public ExpireSnapshotsActionResult execute() {
+  public ExpireSnapshotsActionResult doExecute() {
+    spark.sparkContext().setJobGroup("EXPIRE", "EXPIRE-SNAPSHOTS", false);
```

Comment:
I wonder if `protected JobGroupInfo jobGroup()` in `BaseSparkAction` would make it clearer to implementers that this info should be generated/supplied. Then `execute` in `BaseSparkAction` can call teh `jobGroup` method and set the context itself. I think that is more clear for non-spark experts.

Reply:
+1 on this idea. I think we should add an abstract method to `BaseSparkAction` like @rymurr suggests:

```
protected abstract JobGroupInfo jobGroup();
```

Then we just need to rename `execute` to `doExecute` in actions and set `jobGroup` in `BaseSparkAction`.

Reply:
We may need my refactoring, though, to fix the hierarchy.

Reply:
Data compaction action may submit multiple jobs (e.g. one job per partition) and we would want to have different descriptions for each job (e.g. include the partition we compact in each). Maybe, the abstract method won't work that well in that case.

@RussellSpitzer, thoughts?

Reply:
I'd probably add the following method to `BaseSparkAction` and make implementations call as needed:

```
protected <T> T withJobGroupInfo(JobGroupInfo info, Runnable action) {
    JobGroupInfo previousInfo = JobGroupUtils.getJobGroupInfo(sparkContext);
    try {
      JobGroupUtils.setJobGroupInfo(sparkContext, info);
      return action.run();
    } finally {
      JobGroupUtils.setJobGroupInfo(sparkContext, previousInfo);
    }
}
```